### ISSUE TITLE
Sample for json editor in development mode(loading each dependent file individually )

### ISSUE THREAD

**Issue by arjunballa** [ajaxorg/ace#3888]
@nightwing can you point me to sample for json editor in development mode(loading minimum dependencies individually)
I use a dependency manager stealjs https://stealjs.com which is like requirejs and support commonjs, requirejs and AMD syntax. When I try to load the minified version I am getting errors as steal is not able to understand the requirejs build minified version. Steal understands the requirejs dependency syntax and builds its own minified version. Also I want to use https://github.com/ajaxorg/ace/pull/2885 json extensions and that is another reason I need to build my own version of minified library based on minimum dependencies required to load JSON editor.

I am stuck and not able to make progress. Can you please help me?

**Comment by nightwing** [comment]
@arjunballa is your code open source, or maybe you have a small example of a page with ace and stealjs that i could use to start debugging this issue?

**Comment by arjunballa** [comment]
@nightwing debugging the loading issue with stealjs is not a blocker for me right now. I am changing ace to support auto-suggestions and I am having hard time loading it in development mode(loading each dependent file individually ).


Can you please point me to a sample page which you use to load ace in development mode using requirejs(I am assuming that is the dependency loader ace uses)? I want to use requirejs and sample page  and test changes to ace to support auto-complete. Once I am done with that I will work on stealjs integration issue.

**Comment by nightwing** [comment]
@arjunballa In development mode we use https://github.com/ajaxorg/ace/blob/master/kitchen-sink.html  

stealjs loading issue seems to be the broken detection of define pattern.
The following module breaks stealjs in production mode
```js
console.log(function(a, e, m) {
    a("./z/y");
});
module.exports = 1
```

**Comment by nightwing** [comment]
I have created a rough draft of using json language service with ace, with webpack https://github.com/nightwing/ace-webpack-demo, i'll try to include this as an official feature in the next release.

**Comment by arjunballa** [comment]
@nightwing Thanks for answering.
Just an update - 

I saw the ace-webpack-demo. I didn't get a chance to test it yet. I will test and keep you posted.

For testing my changes to JSON auto-completer I am still using `node Makefile.dryice.js` to build every time I make a change. But it builds in like couple of seconds it is not blocker for me now. 

stealjs detects define pattern and understands it. In the past I loaded many libraries which have define statements in it and it worked. I am not sure why it is not able to load the minified version of files.

I did try to load ace JSON editor individual dependencies using `stealjs` instead of `requires` it works but with one little issue.
steal when looks at  `require('test.css')` it tries to load `test.css.js`. steal has a `!` convention to load load css file like `steal('test.css!')`. I tried `require('test.css!')` and it was not working. Anyways `require('test.css')` is not a permanent solution. I commented all the  css require statements and it worked.

steal detects below define pattern and works fine. 
```
define(function(require, exports, module) {
var abc;
// ...
module.exports = abc;
});
```
